code/modules/wiremod/components/ntnet/ntnet_reciever.dm

```diff
@@ -0,0 +1,63 @@
+/**
+ * # NTNet Reciever Component
+ *
+ * Recieves data through NTNet.
+ */
+
+/obj/item/circuit_component/ntnet_receive
+	display_name = "NTNet Receiver"
+	desc = "Recieves data packages through NTNet. If Encryption Key is set then only signals with the same Encryption Key will be received."
+
+	circuit_flags = CIRCUIT_FLAG_OUTPUT_SIGNAL //trigger_output
+
+	network_id = __NETWORK_CIRCUITS
+
+	var/datum/port/input/push_hid
+	var/datum/port/output/hid
+	var/datum/port/output/data_package
+	var/datum/port/output/secondary_package
+	var/datum/port/input/enc_key
+	var/datum/port/input/option/data_type_options
+
+/obj/item/circuit_component/ntnet_receive/Initialize()
+	. = ..()
+	AddComponent(/datum/component/ntnet_interface)
+	push_hid = add_input_port("Get Hardware ID", PORT_TYPE_SIGNAL)
```

Comment:
You still shouldn't need this, remember--circuits should specify their own "scope", receivers have their own "scope", if a packet is sent with that scope then the receiver fires a signal with the data.